src/DataViews/include/DataViews/DataView.h

```diff
@@ -189,6 +189,7 @@ class DataView {
   void OnCopySelectionRequested(const std::vector<int>& selection);
   void OnExportToCsvRequested();
   virtual void OnExportEventsToCsvRequested(const std::vector<int>& /*selection*/) {}
+  [[nodiscard]] virtual uint64_t GetScopeId(uint32_t row) const;
```

Comment:
I forgot when we change the `indices_` to be the list of scope id (I found a comment there). But this is not true for the `PresetsDataView` and `TracepointsDataView`.

Reply:
I prefer to not assume the `indices_` are scope ids and get the scope id in a similar way as `GetModuleDataFromRow` or `GetFunctionInfoFromRow`.

We can have `GetScopeIdFromRow` and override it in the views where the `indices_` contains scope ids.

Reply:
Just double checked. I think now the `indices_` is used in an inconsistent way in different views.

The original purpose of `indices_` is to find the data that associated with the row. Take `TracepointsDataView` as example, `tracepoints_` is the data, and `tracepoints_[indices_[row]]` is the tracepoint that `row` corresponds to.
Similar for `PresetsDataView`, we have `presets_`. `indices_[row]` is the index in `presets_`.

But in ModulesDataView, we push `start_address` into `indices_` and use `start_address_to_module_.at(indices_[row])` to find the module that the `row` corresponds to.

Reply:
As discussed offline, an abstract class ScopeDataView shall be added, which `LiveFunctionsDataView` and `FunctionsDataView` shall inherit from.